src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -1,181 +1,173 @@
-    });
-  }, [searchQuery, venue, venueEvents, worldUsers]);
+  }, [searchQuery, venue, clearSearchQuery]);
+
+  const foundUsers = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) return [];
+
+    return worldUsers
+      .filter((user) => user.partyName?.toLowerCase().includes(searchQuery))
+      .map((user) => (
+        <NavSearchResult
+          key={`user-${user.id}`}
+          title={user.partyName ?? DEFAULT_PARTY_NAME}
+          image={user.pictureUrl}
+          isAvatar={true}
+          onClick={() => {
+            openUserProfileModal(user);
+            clearSearchQuery();
+          }}
+        />
+      ));
+  }, [searchQuery, worldUsers, clearSearchQuery, openUserProfileModal]);
+
+  const foundEvents = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) return [];
+
+    return venueEvents
+      .filter((event) => event.name.toLowerCase().includes(searchQuery))
+      .map((event) => {
+        const imageUrl =
+          venue?.rooms?.find((room) => room.title === event.room)?.image_url ??
+          venue?.host?.icon;
+        return (
+          <NavSearchResult
+            key={`event-${event.id ?? event.name}`}
+            title={event.name}
+            description={`Event - ${uppercaseFirstChar(
+              formatUtcSecondsRelativeToToday(event.start_utc_seconds)
+            )}`}
+            image={imageUrl}
+          />
+        );
+      });
+  }, [searchQuery, venueEvents, venue]);
 
   const numberOfSearchResults =
-    searchResult.rooms.length +
-    searchResult.events.length +
-    searchResult.users.length;
-
-  const clearSearchQuery = useCallback(() => {
-    setSearchQuery("");
-  }, []);
+    foundRooms.length + foundEvents.length + foundUsers.length;
 
   const clearSearchIcon = (
     <img
-      className="nav__clear-search"
-      src="/icons/nav-dropdown-close.png"
+      className="NavSearchBar__clear-search"
+      src={CloseIcon}
       alt="close button"
       onClick={clearSearchQuery}
     />
   );
 
+  const navDropdownClassnames = classNames("NavSearchBar__nav-dropdown", {
+    "NavSearchBar__nav-dropdown--show": searchQuery,
+  });
+
   return (
-    <div className="nav-search-links">
+    <div className="NavSearchBar">
+      <div className={navDropdownClassnames}>
+        <div className="NavSearchBar__nav-dropdown__title font-size--small">
+          <b className="NavSearchBar__search-results-number">
+            {numberOfSearchResults}
+          </b>{" "}
+          search results
+        </div>
+
+        <div className="NavSearchBar__search-results">
+          {foundRooms}
+          {foundEvents}
+          {foundUsers}
+        </div>
+      </div>
+
       <InputField
-        className="nav__search"
-        value={searchQuery}
-        onChange={onSearchInputChange}
+        inputClassName="NavSearchBar__search-input"
+        onChange={(e: ChangeEvent<HTMLInputElement>) =>
+          debouncedSearch(e.target.value)
+        }
```

Comment:
If we refactor `debouncedSearch` to accept the `ChangeEvent` directly rather than just a string, then we can remove the inline function handler here (which will cause re-renders)

```suggestion
        onChange={debouncedSearch}
```

Reply:
fixed: https://github.com/sparkletown/sparkle/pull/1328/commits/c7104e118d01c1d1d5100c9cf01154be7115c42e